src/components/pages/Reserve.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import Swal from 'sweetalert2';
+import { selectUser } from '../../redux/usersession/usersessionsSlice';
+import { createReservation } from '../../redux/reservations/reservationsSlice';
+
+function Reserve() {
+  const [boats, setBoats] = useState([]);
+  const user = useSelector(selectUser);
+  const [username, setUsername] = useState(user.name);
+  const [city, setCity] = useState('');
+  const [date, setDate] = useState('');
+  const [boat_id, setBoatId] = useState(undefined);
```

Comment:
I noticed a naming convention issue, for useState it's mandatory to have both the variable and the setter to use camelcase, so const [boatId, setBoatId] = useState(undefined);

Please change it here, and 2 other places: line 24 and line 90